File: enterprise_workflow_system/mobile-app/ios/WorkflowiOS/Data/MockData.swift
import Foundation

struct MockData {
    static let userName = "Wei Chen"
    static let userRole = "Senior Developer"
    static let userEmail = "wei.chen@company.com"
    static let userDepartment = "Engineering"

    static let workflowRequests: [WorkflowRequest] = [
        .init(id: "WF-1289", title: "Leave Request — Annual Leave Dec 20-31", type: "Leave", status: .inProgress, date: "Today at 09:15", amount: "", submitter: "Wei Chen", department: "Engineering"),
        .init(id: "PO-8821", title: "Purchase Order — Office Chairs x20", type: "Purchase", status: .pending, date: "Today at 08:30", amount: "¥12,000", submitter: "Wei Chen", department: "Operations"),
        .init(id: "TR-4402", title: "Travel Approval — Beijing Trip", type: "Travel", status: .approved, date: "Yesterday at 14:00", amount: "¥12,500", submitter: "Sarah Tan", department: "Sales"),
        .init(id: "ER-9912", title: "Expense Report — Q3 Dinner", type: "Expense", status: .rejected, date: "Yesterday at 10:45", amount: "", submitter: "John Doe", department: "HR"),
        .init(id: "WF-1240", title: "Equipment Upgrade — Macbook Pro", type: "Equipment", status: .inProgress, date: "3 days ago", amount: "¥18,000", submitter: "Wei Chen", department: "Engineering"),
        .init(id: "WF-1215", title: "Software License — Adobe CC", type: "Purchase", status: .approved, date: "5 days ago", amount: "¥5,400", submitter: "Wei Chen", department: "Engineering"),
    ]

    static let approvalItems: [ApprovalItem] = [
        .init(id: "WF-1234", title: "Equipment Purchase — CNC Machine", description: "Industrial-grade CNC machine", requester: "Wei Chen", department: "Engineering", amount: "¥45,000", priority: "High", date: "2h ago"),
        .init(id: "WF-1235", title: "Travel Approval — Beijing Trip", description: "Q3 business development", requester: "Sarah Tan", department: "Sales", amount: "¥12,500", priority: "Medium", date: "5h ago"),
        .init(id: "WF-1236", title: "Leave Request — Annual Leave", description: "Dec 20-31", requester: "John Doe", department: "HR", amount: "—", priority: "Low", date: "Yesterday"),
        .init(id: "WF-1238", title: "Office Renovation Budget", description: "Floor 3 renovation", requester: "Admin", department: "Operations", amount: "¥150,000", priority: "Urgent", date: "15m ago"),
    ]

    static let notifications: [NotificationItem] = [
        .init(id: "N1", title: "Request Approved", body: "Your leave request #WF-1289 has been approved by Li Wei", type: "approved", time: "5 min ago", isNew: true),
        .init(id: "N2", title: "Approval Required", body: "Equipment Purchase #WF-1234 needs your approval. Amount: ¥45,000", type: "approval_required", time: "23 min ago", isNew: true),
        .init(id: "N3", title: "Escalation Notice", body: "Travel Approval #WF-1235 has been escalated due to overdue deadline", type: "escalation", time: "1h ago", isNew: true),
        .init(id: "N4", title: "Reminder", body: "Purchase Order #WF-1267 is awaiting your response for 24h", type: "reminder", time: "Yesterday", isNew: false),
        .init(id: "N5", title: "Request Completed", body: "Equipment Purchase #WF-1200 workflow completed", type: "completed", time: "Yesterday", isNew: false),
        .init(id: "N6", title: "New Comment", body: "Li Wei commented on #WF-1234: 'Please provide the updated quote'", type: "comment", time: "2 days ago", isNew: false),
        .init(id: "N7", title: "Request Rejected", body: "Expense Report #WF-1210 rejected by Zhang Min", type: "rejected", time: "3 days ago", isNew: false),
    ]
}

File: enterprise_workflow_system/mobile-app/android/app/src/main/kotlin/com/workflow/mobile/data/MockData.kt
package com.workflow.mobile.data

data class Workflow(
    val id: String,
    val title: String,
    val type: String,
    val status: String,
    val date: String,
    val amount: String = "",
    val submitter: String = "Wei Chen",
    val department: String = "Engineering"
)

data class Approval(
    val id: String,
    val title: String,
    val description: String,
    val requester: String,
    val department: String,
    val amount: String,
    val priority: String,
    val date: String
)

data class Notification(
    val id: String,
    val title: String,
    val body: String,
    val type: String, // approved, approval_required, escalation, reminder, completed, comment, rejected
    val time: String,
    val isNew: Boolean
)

data class StatCard(
    val label: String,
    val value: String,
    val color: Long
)

object MockData {
    val userName = "Wei Chen"
    val userRole = "Senior Developer"
    val userEmail = "wei.chen@company.com"
    val userDepartment = "Engineering"

    val stats = listOf(
        StatCard("Pending", "3", 0xFF8B5000),
        StatCard("Active", "7", 0xFF0060A9),
        StatCard("Approved", "28", 0xFF67C23A),
        StatCard("Avg Time", "4.2h", 0xFF191C1E),
    )

    val workflows = listOf(
        Workflow("WF-1289", "Leave Request — Annual Leave Dec 20-31", "Leave", "In Progress", "Today at 09:15", "", "Wei Chen", "Engineering"),
        Workflow("PO-8821", "Purchase Order — Office Chairs x20", "Purchase", "Pending", "Today at 08:30", "¥12,000", "Wei Chen", "Operations"),
        Workflow("TR-4402", "Travel Approval — Beijing Trip", "Travel", "Approved", "Yesterday at 14:00", "¥12,500", "Sarah Tan", "Sales"),
        Workflow("ER-9912", "Expense Report — Q3 Dinner", "Expense", "Rejected", "Yesterday at 10:45", "", "John Doe", "HR"),
        Workflow("WF-1240", "Equipment Upgrade — Macbook Pro", "Equipment", "In Progress", "3 days ago", "¥18,000", "Wei Chen", "Engineering"),
        Workflow("WF-1232", "Policy Exception — Remote Access", "General", "Cancelled", "4 days ago", "", "Wei Chen", "Engineering"),
        Workflow("WF-1215", "Software License — Adobe CC", "Purchase", "Approved", "5 days ago", "¥5,400", "Wei Chen", "Engineering"),
    )

    val approvals = listOf(
        Approval("WF-1234", "Equipment Purchase — CNC Machine", "Industrial-grade CNC machine for precision prototyping", "Wei Chen", "Engineering", "¥45,000", "High", "2h ago"),
        Approval("WF-1235", "Travel Approval — Beijing Trip", "Q3 business development trip", "Sarah Tan", "Sales", "¥12,500", "Medium", "5h ago"),
        Approval("WF-1236", "Leave Request — Annual Leave", "Dec 20-31 annual leave", "John Doe", "HR", "—", "Low", "Yesterday"),
        Approval("WF-1238", "Office Renovation Budget", "Floor 3 renovation proposal", "Admin", "Operations", "¥150,000", "Urgent", "15m ago"),
    )

    val notifications = listOf(
        Notification("N1", "Request Approved", "Your leave request #WF-1289 has been approved by Li Wei", "approved", "5 min ago", true),
        Notification("N2", "Approval Required", "Equipment Purchase #WF-1234 needs your approval. Amount: ¥45,000", "approval_required", "23 min ago", true),
        Notification("N3", "Escalation Notice", "Travel Approval #WF-1235 has been escalated due to overdue deadline", "escalation", "1h ago", true),
        Notification("N4", "Reminder", "Purchase Order #WF-1267 is awaiting your response for 24h", "reminder", "Yesterday", false),
        Notification("N5", "Request Completed", "Equipment Purchase #WF-1200 workflow completed", "completed", "Yesterday", false),
        Notification("N6", "New Comment", "Li Wei commented on #WF-1234: 'Please provide the updated quote'", "comment", "2 days ago", false),
        Notification("N7", "Request Rejected", "Expense Report #WF-1210 rejected by Zhang Min", "rejected", "3 days ago", false),
    )

    val pendingCount get() = approvals.size
    val approvedCount get() = workflows.count { it.status == "Approved" }
    val rejectedCount get() = workflows.count { it.status == "Rejected" }
}
